cmd/porter/credentials.go

```diff
@@ -214,3 +215,28 @@ func buildCredentialsShowCommand(p *porter.Porter) *cobra.Command {
 
 	return cmd
 }
+
+func buildCredentialsCreateCommand(p *porter.Porter) *cobra.Command {
+	opts := porter.CredentialCreateOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create a Credential",
+		Long:  "Create a new blank resource for the definition of a Credential Set.",
+		Example: `
+		porter credentials create FILE [--output yaml|json]
+		porter credentials create credential-set.json
+		porter credentials create credential-set --output yaml`,
```

Comment:
If we are discarding the possibilities of input with `porter credentials create credName --output outputType`, what will be the scenario of the user using the `--output` flag? Since the extension can be parsed from the `FILE` input (`FILE.{json|yaml}`) and which I assume will be the default usecase of the command.

Reply:
Since you mentioned that the default format will be `yaml`, will `porter credentials create creds` save a `creds.yaml` file to disk or return an error since the user didn't specify any format on the command input?

Reply:
Sorry my feedback was a bit piecemeal and I didn't collect it all together to give a single recommendation about how the command should look.

What I am suggesting is the following:

```
porter credentials create [FILENAME] [--output OUTPUT]
```

* When a filename is not specified, we write to standard out in specified output.
* When --output is not specified, and the filename is set, try to use the file extension.
* When --output is not specified, and the filename is not yet, use yaml (and write to stdout).

Here are some examples showing how it looks
```
$ porter credentials create
schemaVersion: 1.0.0
name: NAME
# Omit the namespace to define the credential set in the current current namespace
namespace: NAMESPACE
# Optionally define labels which can be used for filtering. Labels must be string values.
labels:
    LABEL: "LABEL_VALUE"
credentials:
- name: CREDENTIAL_NAME
  source:
    # Replace the value field below with any of the following: secret, env, path, command or value
    value: CREDENTIAL_VALUE

$ porter credentials create mycreds.yaml
# writes yaml formatted creds to mycreds.yaml

$ porter credentials create mycreds.yml
# writes yaml formatted creds to mycreds.yml

$ porter credentials create mycreds.json
# writes json formatted creds to mycreds.json

$ porter credentials create creds
could not detect the file format from the file extension (.txt). Specify the format with --output.

$ porter credentials create creds --output json
# writes json formatted creds to mycreds
```

Reply:
Ah sorry, I should be the one listing all the possibilities of the command. Thank you for doing it, Carolyn 🙂 

> I am a bit concerned about this use case here, where they specify the file name but not the extension. It doesn't quite fit the existing behavior of porter and could easily result in people misunderstanding how the command works and wondering where the file went.
> 
> Let's remove this logic and just have Porter use the filename that they pass in as an argument without modification. It will be easier for users to predict what porter will do and in general one of our design principle's is to not surprise the user.

I apologize in advance if I still don't fully get it. On the first comment you reference to the use case of use specifying file name without the extension (`porter credentials create credential-set --output yaml` on my code) and said to remove the logic, but you mentioned the similar scenario on the last example on your list (`porter credentials create creds --output json` on your code snippet). How are those two different and which logic were you referring to that should be removed?

Reply:
In all examples the positional argument is a file path. `porter credentials create creds -o json` should be interpreted as "generate the credentials in json format and save to a file named creds (without any extension)". I used that example specifically to help clarify that we shouldn't assert in code that the file extension MUST match the --output flag.

When no --output is specified, we try to guess the format from the file extension. If we can't we error out and tell them to provide --output (see second to last example). 

Does that help clarify? The problem I have with what is currently implemented is that we enforce that the --output and the file extension match.

Reply:
I literally had my eureka moment and said "OOH" out loud after understanding what you meant. First and foremost, I am really sorry that it took me some time get it and sorry if it frustrated you that you need to explain these multiple times. Thank you so much for the patience, @carolynvs ! Really aprpeciate it. 🙂 

There are still few things that I want to clarify before proceeding with the fixes:
1. At the examples that you listed above, you mentioned that the command of `porter credentials create creds` would yield error since we can't deduce the output format both from the filename because it didn't supply the extension and also because there is no `--output` flag input. But since you suggested [here](https://github.com/getporter/porter/pull/1838#discussion_r754508807) to use the default value of `yaml` for the `--output` flag, shouldn't the command `porter credentials create creds` then write the credentials in `yaml` format and save it to a file in disk name `creds` since the output flag with no input will automatically be filled with `yaml` as the flag default value?
2. As you mentioned [here](https://github.com/getporter/porter/pull/1838#discussion_r754503628) to use the `switch` primitive for checking the file format and since the `Marshal` example also mentioned it, do we also want the `.toml` file as one of the option for the `--output` format?

Reply:
I actually suggested this when I was saying use yaml as a default. Note that no file was passed, so we are just printing to the console and not saving to a file.

```
$ porter credentials create
# defaults --output to yaml and outputs it to stdout
```

Maybe I've made this too confusing and users would have trouble picking up on the subtle differences between the ways to call this command though... 🤔 Like we could just not help write the file at all and say people have to use the redirect operators like `porter credentials create --output json > mycreds.json`

Reply:
@carolynvs The command example that you mentioned of having no file input passed (`porter credentials create`) was numbered 1 on my screenshot below while the one that I was referring was the command number 2 (`porter credentials create creds`), where user inputs the file (`creds`). 

![image](https://user-images.githubusercontent.com/7043511/150617463-dca49799-0fbe-4139-9985-f11468d8dee7.png)

It was mentioned that it should have yield error since we couldn't deduce the file format from either of the file input or the `--output` flag. But shouldn't it write YAML format output to the `creds` file at the disk since the `--output` flag will have a default value of YAML?

Reply:
Sometimes the default value for a flag isn't a static value, and that's what I was suggesting. i.e. instead of setting the output default directly with cobra when we define the flag, the Validate function for the create options struct should default it with more complex logic.

1. When no filename is specified, default output to yaml.
2. When a file name is specified, default output using the file extension.